data/domain/player.tsx

```diff
@@ -728,14 +729,24 @@ export const updatePlayerStarSigns = (data: Map<string, any>) => {
     })
 }
 
+export const updatePlayerMeditating = (data: Map<string, any>) => {
+    const players = data.get("players") as Player[];
+    const divinityData = data.get("divinity") as Divinity;
+
+    players.forEach(player => {
+        const playerDivinityData = divinityData.playerInfo[player.playerID];
+        player.meditating = (player.getActivityType() == Activity.Divinity || player.getActivityType() == Activity.Lab && playerDivinityData.gods.some(god => god.data.name == 'Goharut'));
```

Comment:
Yeah this is much easier to update / use in other place in my opinion, good job!